### DISCUSSION TITLE
Problem sa prikazom temprerature, vlaznosti i pritiska vazduha

### DISCUSSION THREAD

**Discussion by donpedjoni78** [DesconBelgrade/Klimerko#41] [General]
Klimerko ne ppokazuje vrednosti temperature, pritiska i vlaznosti i desava se da se diskonektuje sa SBB rutera.
Vrednost temperature prikazuje kao -3.99c, a vlaznost 0% kao i 0mba.
Firmware Version  2.1.0

**Comment by epiller** [comment]
Djeluje mi da su to dva odvojena problema.  
Što se tiče temperature, vlažnosti, i pritiska:
- Jeste li sigurno dobro spojili BME280 senzor?
- Da li ste sigurni da koristite [BME280 senzor a ne BMP280?](https://i.imgur.com/jnMkxKy_d.webp)?  
  
Što se tiče rutera, tu ne mogu mnogo pomoći jer postoji previše varijabli... Pošaljite debug output sa serijskog porta Klimerka gdje se vidi trenutak kada pukne veza sa ruterom.

**Reply by donpedjoni78** [reply]
Klimerko nisam sastavljao vec dobio u onoj akciji od prošle godine i radio je.  Testiram ga na MTS ruteru i za sada radi nekoliko sati.
Mogu da probam da vratim stariji firmware.

**Comment by donpedjoni78** [comment]
![image](https://user-images.githubusercontent.com/80104413/202904446-68fb5a92-1a2f-408a-8410-3e4d60c6039f.png)

**Reply by epiller** [reply]
Pogledajte [uputstvo](https://github.com/DesconBelgrade/Klimerko#bme280-sensor-preparation) i provjerite da li su sve žice spojene kako treba. Moguće je i da se BME280 senzor pokvario.

**Reply by donpedjoni78** [reply]
Nisam primetio da se neka zica odlemila. Ova verzija klimerka je malo drugacije napravljena nego sto si ti pokazao na snimku.
Narucicu sa Aliexpressa nov senzor. Vidim da postoje dve verzije, jedna je 3.3V, a druga 5V. Potrebna je ova od 5V?

![image](https://user-images.githubusercontent.com/80104413/202908511-32a03ddf-5d4d-4b10-a0cb-78580e76c2d8.png)

**Reply by epiller** [reply]
Kupite 3.3v verziju, iako mislim da i 5v verzija radi sa 3.3v.  
U [Hardware Required](https://github.com/DesconBelgrade/Klimerko#hardware-required) dijelu dokumentacije sam otprilike objasnio šta treba da se kupi.

**Reply by donpedjoni78** [reply]
Na mom Klimerku je upravo ovakva verzija. Na Aliexpressu verzija od 3.3V izgleda ovako:

![image](https://user-images.githubusercontent.com/80104413/202909671-7d64125c-805a-454d-8ac5-43395e3e83f9.png)

**Reply by epiller** [reply]
I jedna i druga verzija bi trebala da radi bez problema, ali nabavite 3.3v ako je jeftinija.

**Reply by donpedjoni78** [reply]
Hvala. Narucicu ovu verziju koja je kod mene. Da li moze na neki nacin da se testira senzor van Klimerka?

**Reply by epiller** [reply]
NodeMCU (ESP8266) je mikrokontroler na kom se Klimerko bazira, tako da možete napisati neki drugi firmware koji će služiti za testiranje BME senzora. [Evo pomoć](https://randomnerdtutorials.com/bme280-sensor-arduino-pressure-temperature-humidity/).

**Reply by donpedjoni78** [reply]
Hvala. Senzor se ne odaziva. Verovatno je pokvaren.

**Reply by donpedjoni78** [reply]
![image](https://user-images.githubusercontent.com/80104413/202913977-d705bd32-44c8-496d-8e89-47a40eceed55.png)

Cudno se ponasa. Uspeo sam da dobijem vrednosti na allthingstalk, ali na Serial Monitoru i dalje izbacuje :
Temperature/Humidity/Pressure Sensor (BME280) returned no data this time.

**Reply by epiller** [reply]
Izgleda da nekada uspije da očita vrijednosti pa ih smjesti u rolling average i onda tu vrijednost pošalje. To može biti jednom u npr. 100 pokušaja, što je dovoljno da se vrijednost prikaže na AllThingsTalk Maker-u a nedovoljno da bi se primjetilo u Serial Monitor-u (osim ako se ne obraća pažnja na svako očitavanje).

**Reply by vultureEmb** [reply]
Ja imam problem sa BME senzorom sa 6 pinova, maši temperaturu za 5 stepeni, ne znam da li se radi neka kalibracija senzora na početku ili? Bilo bi dobro dodati NO2, SO2 i O3 senzor unutar kucista, trebalo bi da moze da stane?

**Reply by epiller** [reply]
@vultureEmb Updatuj uredjaj na [Firmware v2.1.0](https://github.com/DesconBelgrade/Klimerko/releases/tag/2.1.0), koji riješava taj problem tako što ti daje opciju mijenjanja temperature offset-a unutar WiFi Configuration Portal-a  😄  
Što se tiče NO2 i SO2 senzora, postoji [Klimerko Pro](https://github.com/isocserbia/Klimerko-Pro) projekat koji sadrži upravo te senzore, ali su dosta skupi...

**Reply by vultureEmb** [reply]
@epiller moram da primetim da je BME280 dosta neprecizan po pitanju temperature, -4 offset ne rešava problem, ne znam da li postoji neka alternativa preciznija, pa da se doradi za njega firmware.

**Reply by Buljubasa** [reply]
BME280 je dobar senzor ali pored toga sto inace pokazuje nesto visu temperaturu (obicno  do 1C) ovde se nalazi na vrhu kucista tako da sva toplota koju disipira elektronika ispod dodatno podize tu temperaturu ali ne bas linearno, dosta zavisi od spoljne temperature.
Ja sam to resio tako sto sam napravio otvor ispod PMS senzora u koji sam smestio BME280 tako da svez vazduh prolazi prvo preko njega i pokazivanje je mnogo realnije.

**Reply by Buljubasa** [reply]
A sto se tice pokazivanja sa pocetka posta, imam dva klimerka vec neko vreme i ono sto se meni desilo je da kako uredjaji stoje napolju ume da se pojavi oksidacija, paucina, prasina itd... izmedju pinova na bme senzoru kao i nodemcu koja prouzrokuje lose ili nikakvo ocitavanje a jedan mi je i preminuo.
Pogledati povremeno u kakvom su stanju i ocistiti ih i prelemiti ako za tim ima potrebe.
Oko PMS7003 senzora sam imao problem samo kada pad napona od ispravljaca bude primetan i padne ispod 4.5V (napon meriti kada radi njegov ventilator).
Recimo na 4V senzor radi ali pokazuje krajnje neprecizno. To se resava kvalitetnim kablom i sto manjom duzinom istog.

**Reply by vultureEmb** [reply]
> BME280 je dobar senzor ali pored toga sto inace pokazuje nesto visu temperaturu (obicno do 1C) ovde se nalazi na vrhu kucista tako da sva toplota koju disipira elektronika ispod dodatno podize tu temperaturu ali ne bas linearno, dosta zavisi od spoljne temperature. Ja sam to resio tako sto sam napravio otvor ispod PMS senzora u koji sam smestio BME280 tako da svez vazduh prolazi prvo preko njega i pokazivanje je mnogo realnije.

Za poziciju senzora si u pravu svakako, mislim da postavljanje senzora u kuciste nije dobro resenje, jer ostale plocice bar neku toplotu otpustaju, plus je kao sto kazes senzor u vrhu kucista. Bilo bi dobro npr kad bi neka sonda mogla da se izbaci napolje i preko nje meri temperatura a sama BME plocica da ostane unutra.

**Reply by Buljubasa** [reply]
> > Bilo bi dobro npr kad bi neka sonda mogla da se izbaci napolje i preko nje meri temperatura a sama BME plocica da ostane unutra.

Ja sam celu plocicu senzora kroz otvor izbacio tako da polovina plocice sa sve senzorom viri van kucista sa donje strane a drugi sam promenio malo tako da ima svoje mesto.

![image](https://user-images.githubusercontent.com/97301975/221169458-106c0b16-e534-4ca5-bb4d-da2cc6d0ad8e.png)
